builtins/web/fetch/request-response.cpp

```diff
@@ -601,6 +606,43 @@ bool RequestOrResponse::parse_body(JSContext *cx, JS::HandleObject self, JS::Uni
     }
 
     result.setObject(*blob);
+
+  } else if constexpr (result_type == RequestOrResponse::BodyReadResult::FormData) {
+    RootedObject headers(cx, RequestOrResponse::maybe_headers(self));
+    if (!headers) {
+      api::throw_error(cx, FetchErrors::InvalidFormDataHeader);
+      return RejectPromiseWithPendingError(cx, result_promise);
+    }
+
+    auto content_type_str = host_api::HostString("Content-Type");
+    auto idx = Headers::lookup(cx, headers, content_type_str);
+    if (!idx) {
+      api::throw_error(cx, FetchErrors::InvalidFormDataHeader);
+      return RejectPromiseWithPendingError(cx, result_promise);
+    }
+
+    auto values = Headers::get_index(cx, headers, idx.value());
+    auto maybe_mime = extract_mime_type(std::get<1>(*values));
+    if (!maybe_mime) {
+      api::throw_error(cx, FetchErrors::InvalidFormDataHeader);
+      return RejectPromiseWithPendingError(cx, result_promise);
+    }
+
+    auto parser = FormDataParser::create(maybe_mime.value().to_string());
+    if (!parser) {
+      return JS::ResolvePromise(cx, result_promise, result);
+      // TODO: add parser for url/encoded data
+      // return RejectPromiseWithPendingError(cx, result_promise);
```

Comment:
This seems like it shouldn't land as-is, and I assume is preparation for #208? If so, could you make sure that it does the right thing in the meantime? Resolving a promise instead of rejecting it with an error doesn't seem right here.